app/models/batch_process.rb

```diff
@@ -121,6 +148,12 @@ def refresh_metadata_cloud_csv
     create_parent_objects_from_oids(oids, metadata_sources, admin_sets)
   end
 
+  def remove_from_metadata_cloud_csv
+    metadata_sources = parsed_csv.entries.map { |r| r['source'] }
```

Comment:
its expensive to run through these columns twice. can we do a single pass instead?